File: 프로그래머스/0/120923. 연속된 수의 합/연속된 수의 합.swift
import Foundation

func solution(_ num:Int, _ total:Int) -> [Int] {
    var result: [Int] = []
    var a1 = (total / num) + (1 - num) / 2

    for _ in 0..<num {
        result.append(a1)
        a1 += 1
    }
    return result
}

File: 프로그래머스/0/120923. 연속된 수의 합/연속된 수의 합.cpp
#include <algorithm>
#include <iostream>
#include <string>
#include <vector>

using namespace std;

vector<int> solution(int num, int total) {
    vector<int> answer;
    int start = 0, end = 0, psum = 0, size = 1;

    while (true) {
        if (psum == total && size == num) {
            break;
        }
        if (psum < total) {
            end++;
            psum += end;
            if (size != num) {
                size++;
            } else {
                psum -= start;
                start++;
            }
        } else {  // psum > total
            start--;
            psum += start;
            if (size != num) {
                size++;
            } else {
                psum -= end;
                end--;
            }
        }
    }

    for (int i = start; i <= end; i++) {
        answer.push_back(i);
    }

    return answer;
}